### PR TITLE
test: Migrate CI to GitHub Actions [SDK-4452]

### PR DESCRIPTION
**Pull request by evansims**
By submitting a PR to this repository, you agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md). Please see the [contributing guidelines](https://github.com/auth0/.github/blob/master/CONTRIBUTING.md) for how to create and submit a high-quality PR for this repo.

### Description

This pull request replaces our current CircleCI continuous integration suite for auth0-react with a new workflow using GitHub Actions.

### References

Please see internal ticket SDK-4452.

### Testing

This pull request replaces our continuous integration suite with one built in GitHub's native Actions ecosystem. It aligns with other migrations taking place across our repositories.

### Checklist

- [x] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not the default branch

### CODE REVIEW

**Comment by evansims** [comment]
@adamjmcgrath I've updated the publish step to only reference the `express-oauth2-jwt-bearer` path.

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        type: string
+      dry-run:
+        type: boolean
+        description: Perform a publishing dry run. This will not publish the release, but will validate the release and log the commands that would be run.
+        default: false
+
+permissions:
+  contents: read
+  id-token: write # For publishing to NPM with provenance. Allows developers to run `npm audit signatures` and verify release signature of SDK. @see https://github.blog/2023-04-19-introducing-npm-package-provenance/
+  packages: write # For cross-publishing to GitHub Packages registry.
```
Comment (frederikprijck) [inline .github/workflows/publish.yml:23]:
```suggestion
```

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+          cache: npm
+          registry-url: "https://registry.npmjs.org"
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Publish release to NPM
```
Comment (frederikprijck) [inline .github/workflows/publish.yml:107]: I think we should run `npm run build` prior to publishing, same as we currently do on circle: https://github.com/auth0/node-oauth2-jwt-bearer/blob/main/.circleci/config.yml#L22

**Review by frederikprijck — CHANGES_REQUESTED** [review]

**Comment by evansims** [comment]
@frederikprijck Updated, thanks.

**Review by frederikprijck — APPROVED** [review]

**Pull request merged**